### PR TITLE
Don't Sleep(1) in some spin-wait loops (Fx)

### PR DESCRIPTION
**Pull request by kouvel**
Fx counterpart to https://github.com/dotnet/coreclr/pull/21722

### CODE REVIEW

**Comment by kouvel** [comment]
@dotnet-bot test Windows x64 Debug Build

File: src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs [hunk 1]
```diff
@@ -384,7 +384,7 @@ private void PushCore(Node head, Node tail)
             // Keep trying to CAS the existing head with the new node until we succeed.
             do
             {
-                spin.SpinOnce();
+                spin.SpinOnce(sleep1Threshold: -1);
```
Comment (stephentoub) [inline src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs]: I'm a little surprised the SpinOnce call here is valuable.  This is one of the ones you tried removing and saw a regression as a result?

Comment (kouvel) [reply src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs]: Yea I tried removing this one and the one in Pop, simplifying the one in pop to remove the extra backoff, so far those things didn't pan out, even the extra backoff in Pop seems to be (marginally) better than just one SpinOnce().

**Comment by kouvel** [comment]
@dotnet-bot test Linux arm64 Release Build

**Comment by kouvel** [comment]
@dotnet-bot test Linux arm64 Release Build

**Pull request merged**